tck/src/main/java/ee/jakarta/tck/data/web/validation/ValidationTests.java

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package ee.jakarta.tck.data.web.validation;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.jupiter.api.BeforeEach;
+
+import ee.jakarta.tck.data.framework.junit.anno.AnyEntity;
+import ee.jakarta.tck.data.framework.junit.anno.Assertion;
+import ee.jakarta.tck.data.framework.junit.anno.Web;
+import ee.jakarta.tck.data.framework.utilities.TestPropertyUtility;
+import jakarta.inject.Inject;
+import jakarta.validation.ConstraintViolationException;
+
+@Web
+@AnyEntity
+public class ValidationTests {
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addClasses(Rectangle.class, Rectangles.class);
+    }
+    
+    @Inject
+    private Rectangles rectangles;
+    
+    @BeforeEach
+    public void cleanup() {
+        rectangles.deleteAll();
```

Comment:
Does this also need to wait for eventual consistency?  Some tests count the total number of rectangles so if some from a previous test can sometimes still be seen, it seems like this might cause an intermittent failure.

Reply:
It does for NoSQL, however, we can keep the focus on SQL while we don't have the NoSQL spec yet.

Reply:
Good catch, fixed :)